docs: Adds documentation for image decoding attribute

SkyZeroZx opened the pull request:
Expands the image optimization guide to explain browser image decoding strategies

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/contributing-docs/commit-message-guidelines.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.dev application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: N/A


## What is the new behavior?


## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

JeanMeche commented on adev/src/content/guide/image-optimization.md:
```diff
@@ -38,8 +38,8 @@ and include it into the `imports` array of a standalone component or an NgModule
 <docs-code language="typescript">
 
 imports: [
-  NgOptimizedImage,
-  // ...
+NgOptimizedImage,
+// ...
 ],
 
 </docs-code>
```
````suggestion
```ts

imports: [
  NgOptimizedImage,
  // ...
],

```
````

Let's replace that code block so we don't break again the identation.

SkyZeroZx replied:
Done

JeanMeche commented:
Do you mind also adding a link to that guide to `packages/common/src/directives/ng_optimized_image/ng_optimized_image.ts` with a `@see` tag.

SkyZeroZx commented:
Question : Should we change all the angular html and typescript tags, are we already on it?
like this ? 
```ts
<docs-code language="typescript">
providers: [
  {
    provide: IMAGE_CONFIG,
    useValue: {
      breakpoints: [16, 48, 96, 128, 384, 640, 750, 828, 1080, 1200, 1920]
    }
  },
],
</docs-code>
```

SkyZeroZx commented:
Done

SkyZeroZx commented on adev/src/content/guide/image-optimization.md:
````diff
@@ -303,6 +301,33 @@ By default, `NgOptimizedImage` sets `loading=lazy` for all images that are not m
+You can still override this behavior by explicitly setting the `decoding` attribute.  
+[See the documentation for the standard image `decoding` attribute for details](https://developer.mozilla.org/docs/Web/HTML/Element/img#decoding).
+
+```angular-html
````
Done

SkyZeroZx commented on adev/src/content/guide/image-optimization.md:
```diff
@@ -180,11 +179,11 @@ By default, NgOptimizedImage applies a CSS blur effect to image placeholders. To
+[placeholderConfig]="{blur: false}"
 />
 
 </docs-code>
```
Done

SkyZeroZx committed:
docs: Adds documentation for image decoding attribute

Expands the image optimization guide to explain browser image decoding strategies
